README.rst

```diff
@@ -76,23 +82,50 @@ trailing ``:2.7`` makes sure we are using Python 2.7.
 4. Analysis workflow
 ====================
 
-The hello world example is simple enough in that it does not require any complex
-workflow steps to run. It consists of running a single command only.
-Nevertheless, it is represented in a structured YAML manner with `Yadage
-<https://github.com/diana-hep/yadage>`_ and `Common Workflow Language
-<http://www.commonwl.org/v1.0/>`_ specifications. The corresponding workflow
-descriptions can be found here:
+This analysis is very simple because it consists basically of running only the
+Python executable which will produce a text file.
+
+The workflow can be represented as follows:
+
+.. code-block:: console
+
+                START
+                |
+                |
+                V
+    +-------------------------------+
+    | run Python program            | <-- inputfile
+    |                               | <-- outputfile, sleeptime
+    |    $ python helloworld.py ... |
+    +-------------------------------+
+                |
+                | greetings.txt
+                V
+                STOP
+
+
+Note that you can also use `CWL <http://www.commonwl.org/v1.0/>`_ or `Yadage
+<https://github.com/diana-hep/yadage>`_ workflow specifications:
 
 - `Yadage workflow definition <workflow/yadage/workflow.yaml>`_
 - `CWL workflow definition <workflow/cwl/helloworld.cwl>`_
 
-Local testing with Docker
-=========================
+5. Output results
+-----------------
 
-Let us test whether everything works well locally in our containerised
-environment. We shall use Docker locally. Note how we mount our local
-directories ``inputs``, ``code`` and ``outputs`` into the containerised
-environment:
+The example produces a file greeting all names included in the
+`names.txt <inputs/names.txt>`_ file.
```

Comment:
It would be good to show it inline, just like we do with plots.  Makes example more readable.